### PR TITLE
Add types for lockfile-lint-api v5.1

### PR DESCRIPTION
**Pull request by Snaptags**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [X] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [X] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [X] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [X] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/lockfile-lint-api/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+export interface validationSuccess {
+    type: 'success';
+    object: packages;
+}
+
+export type validationResult = validationError | validationSuccess;
+
+export class ValidateHost {
+    constructor(packages: { packages: packages });
+    validate: (hosts: hosts, options?: validationOptions) => validationResult;
```
Comment (rbuckton) [inline types/lockfile-lint-api/index.d.ts:40]: Are these supposed to be properties, or should they be defined as methods?

```suggestion
    validate(hosts: hosts, options?: validationOptions): validationResult;
```

**Review by rbuckton — CHANGES_REQUESTED** [review]

File: types/lockfile-lint-api/index.d.ts (hunk 1 above)
Comment (Snaptags) [reply types/lockfile-lint-api/index.d.ts]: Thanks for pointing this out, they should indeed be methods.

File: types/lockfile-lint-api/yarn.test.lock [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (andrewbranch) [inline types/lockfile-lint-api/yarn.test.lock:1]: Your test doesn’t actually _execute_, so you should remove this file. The test is just to validate the types that arise from using the library.

**Review by andrewbranch — CHANGES_REQUESTED** [review]

File: types/lockfile-lint-api/yarn.test.lock (hunk 2 above)
Comment (Snaptags) [reply types/lockfile-lint-api/yarn.test.lock]: Thanks, its completely obsolete then, as is the OTHER_FILES.txt. I removed both. Thanks!

**Review by andrewbranch — APPROVED** [review]

**Comment by Snaptags** [comment]
Ready to merge

**Pull request merged**